jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/HttpOutput.java

```diff
@@ -244,8 +244,9 @@ else if (_state == State.CLOSE)
 
         try
         {
-            if (failure != null)
-                _servletChannel.abort(failure);
+            // TODO is this necessary?
+            // if (failure != null)
+            //    _servletChannel.abort(failure);
```

Comment:
@lorban @sbordet I think these lines are not necessary.     Why are we aborting on a write failure AND delivering that failure to the application, which tries to complete, resulting in an exception like:
```
java.lang.IllegalStateException: s=IDLE rs=BLOCKING os=ABORTED is=IDLE awp=false se=false i=true al=0
	at org.eclipse.jetty.ee10.servlet.ServletChannelState.complete(ServletChannelState.java:701)
	at org.eclipse.jetty.ee10.servlet.AsyncContextState.complete(AsyncContextState.java:61)
	at org.eclipse.jetty.ee10.servlets/org.eclipse.jetty.ee10.servlets.EventSourceServlet$EventSourceEmitter.close(EventSourceServlet.java:230)
	at org.eclipse.jetty.ee10.servlets/org.eclipse.jetty.ee10.servlets.EventSourceServlet$EventSourceEmitter.run(EventSourceServlet.java:207)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:577)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1589)
```

Reply:
This seems related to the bug I recently analyzed [here](https://github.com/jetty/jetty.project/issues/10357#issuecomment-1786824936). Or are those two problems distinct?

Reply:
@lorban I think it is very much the same class of problem if not the same problem.    The issue is that the ServletChannel FSM is trying to run the handling to completion, but if some other event forces a completion prior, then that FSM is handling a completed/destroyed/recycled request/response.         The solution is to clean up the error handling to make sure that we do not complete the callback on behalf of the ServletChannel, unless we are sure that it has completed its handling.